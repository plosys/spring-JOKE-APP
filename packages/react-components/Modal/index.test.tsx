import { fireEvent, render } from '@testing-library/react';

import { Modal } from './index.js';

import.meta.jest.mock('react-intl', () => {
  const reactIntl = jest.requireActual('react-intl');
  const intl = reactIntl.createIntl({
    locale: 'en',
  });

  return {
    ...reactIntl,
    useIntl: () => intl,
  };
});

it('should not render a bulma modal when it is inactive', () => {
  const { container } = render(<Modal isActive={false}>test</Modal>);
  expect(container).toMatchSnapshot();
});

it('should render a bulma modal when it is active', () => {
  const { container } = render(<Modal isActive>test</Modal>);
  expect(container).toMatchSnapshot();
});

it('should close the modal when the close button is clicked', () => {
  const onClose = import.meta.jest.fn();
  const { getByLabelText } = render(
    <Modal closeButtonLabel="Test close" isActive onClose={() => onClose()}>
      test
    </Modal>,
  );
  getByLabelText('Test close').click();
  expect(onClose).toHaveBeenCalledWith();
});

it('should close the modal when the background is clicked', () => {
  const onClose = import.meta.jest.fn();
  const { getByRole } = render(
    <Modal isActive onClose={() => onClose()}>
      test
    </Modal>,
  );
  getByRole('presentation').click();
  expect(onClose).toHaveBeenCalledWith();
});

it('should close the modal escape is pressed on the background', () => {
  const onClose = import.meta.jest.fn();
  const { getByRole } = render(
    <Modal isActive onClose={() => onClose()}>
      test
    </Modal>,
  );
  fireEvent.keyDown(getByRole('presentation'), { key: 'Escape' });
  expect(onClose).toHaveBeenCalledWith();
});

it('should not close the modal another key is pressed on the background', () => {
  const onClose = import.meta.jest.fn();
  const { getByRole } = render(
    <Modal isActive onClose={() => onClose()}>
      test
    </Modal>,
  );
  fireEvent.keyDown(getByRole('presentation'));
  expect(onClose).not.toHaveBeenCalled();
});
