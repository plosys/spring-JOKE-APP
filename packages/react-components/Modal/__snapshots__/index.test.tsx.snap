// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render a bulma modal when it is inactive 1`] = `<div />`;

exports[`should render a bulma modal when it is active 1`] = `
<div>
  <div
    class="is-active modal root open"
  >
    <div
      class="modal-background"
      role="presentation"
    />
    <div
      class="modal-content"
    >
      test
    </div>
    <button
      class="modal-close is-large"
      type="button"
    />
  </div>
</div>
`;
